runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -1045,51 +1126,55 @@ func timeRangeClause(tr *runtimev1.TimeRange, mv *runtimev1.MetricsViewSpec, dia
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_COMPARISON_VALUE:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_COMPARISON_VALUE
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_COMPARISON_VALUE
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_ABS_DELTA:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_ABS_DELTA
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_ABS_DELTA
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_REL_DELTA:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_REL_DELTA
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_REL_DELTA
+			}
+		}
+
+		if s.SortType == runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_UNSPECIFIED {
+			return fmt.Errorf("sort measure '%s' does not have a sort type", s.Name)
+		}
+		// check if sorting measure is a derived measure, if it is then it should be present in comparison map
+		if s.SortType != runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE && !comparisonMap[s.Name] {
+			if compareQuery {
+				return fmt.Errorf("sort measure '%s' is not present in the comparison measures", s.Name)
 			}
+			// for backward compatibility, UI uses the old state while switching from compare to no comparison
+			// in that case just sort the measure by base value
+			s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE
 		}
 	}
+	return nil
 }
 
-func validateMeasureAliases(aliases []*runtimev1.MetricsViewComparisonMeasureAlias, hasComparison bool) error {
-	for _, alias := range aliases {
-		switch alias.Type {
-		case runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_COMPARISON_VALUE,
-			runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_ABS_DELTA,
-			runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_REL_DELTA:
-			if !hasComparison {
-				return fmt.Errorf("comparison not enabled for alias %s", alias.Alias)
-			}
+func validateAlias(aliases []*runtimev1.MetricsViewComparisonMeasureAlias, comparisonMap map[string]bool) error {
+	for _, a := range aliases {
+		if a.Type != runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE && !comparisonMap[a.Name] {
+			return fmt.Errorf("alias measure '%s' is not present in the comparison measures", a.Name)
```

Comment:
The old one checked explicitly for the comparison cases. I think that might be better because a) I think we also allow passing `METRICS_VIEW_COMPARISON_MEASURE_TYPE_UNIDENTIFIED`, b) new measure types that are not comparison-related might be added in the future.

Reply:
I commonized common checks for both toplist and compare toplist cases so using the comparisonMap, toplist case will have comparisonMap empty. 
a. Yes it was being passed earlier but the new code disallows that as was not sure whats the use of that and that makes the semantics loose. For example, currently there is a edge case where front end will send a relative sort type for non comparison toplist query, earlier there was no strict check in the toplist logic so it was working, just wanted to avoid similar cases. 
b. May be we use `METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE` for them, if not I can revert it back to what it was previously.

Reply:
reverted the change